### PR TITLE
build(deps): bump github.com/alexkohler/nakedret to 2.0.1

### LINKED ISSUE #1317: nakedret doesn't check "return"s in conditionals
The nakedret linter embedded in golangci-lint appears to only check for "return" statements at the top level of the function, outside of conditionals.  My expectation is that all returns are checked in functions above the specified length, regardless of nesting.

In golangci-lint's test suite, should line 6 of test/testdata/nakedret.go have an "// ERROR" marker?  Letting the first "return" pass here seems wrong:
```
func NakedretIssue() (a int, b string) {
	if a > 0 {
		return
	}

	if b == "" {
		return 0, "0"
	}
[...]
	return // ERROR "naked return in func `NakedretIssue` with 31 lines of code"
}
```

Thanks!
-- Aaron

### PR DESCRIPTION
**Pull request by zalenskivolt**
Handling of naked returns inside conditionals as well as from nested functions has been added upstream, in https://github.com/alexkohler/nakedret/pull/12 and more commits.

This should resolve #1317.

### CODE REVIEW

File: pkg/golinters/nakedret.go [hunk 1]
```diff
@@ -72,63 +73,68 @@ type nakedretVisitor struct {
...
-				if ident != nil {
-					namedReturns = append(namedReturns, ident)
-				}
-			}
+	var (
+		funcType *ast.FuncType
+		funcName string
+	)
+	switch s := node.(type) {
+	case *ast.FuncDecl:
```
Comment (zalenskivolt) [inline pkg/golinters/nakedret.go:104]: `Visit()` from upstream
https://github.com/alexkohler/nakedret/blob/d95c37512/nakedret.go#L196-L235
only updated with the issue reporting.

File: pkg/golinters/nakedret.go [hunk 2]
```diff
@@ -72,63 +73,68 @@ type nakedretVisitor struct {
...
-		if funcDecl.Name == nil {
-			continue
+func hasNamedReturns(funcType *ast.FuncType) bool {
+	if funcType == nil || funcType.Results == nil {
+		return false
+	}
+	for _, field := range funcType.Results.List {
+		for _, ident := range field.Names {
+			if ident != nil {
+				return true
```
Comment (zalenskivolt) [inline pkg/golinters/nakedret.go:91]: `hasNamedReturn` from upstream
https://github.com/alexkohler/nakedret/blob/d95c37512/nakedret.go#L182-L194

**Comment by ldez** [comment]
Hello,

the way that we handle this linter is not right, we have to change that.

Instead of "duplicating" the code inside golangci-lint I think it's better to a create PR on nakedret to be able to use it as a dependency.

**Comment by zalenskivolt** [comment]
> Instead of "duplicating" the code inside golangci-lint I think it's better to a create PR on nakedret to be able to use it as a dependency.

This was updated in https://github.com/alexkohler/nakedret/pull/25 and now it uses that analyzer directly!

**Comment by zalenskivolt** [comment]
@ldez linter is updated, and should not be blocked any more! 😄

**Review by ldez — APPROVED** [review]
LGTM

File: test/testdata/nakedret.go [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
 
+import "fmt"
+
 func NakedretIssue() (a int, b string) {
 	if a > 0 {
-		return
+		return // want "naked return in func `NakedretIssue` with 33 lines of code"
 	}
 
+	fmt.Println("nakedret")
```
Comment (zalenskivolt) [inline test/testdata/nakedret.go:11]: @ldez were these `fmt.Println()` updates intentional?

should it still test the edge case of 31 lines, by removing two `// ...` lines below?

Comment (ldez) [reply test/testdata/nakedret.go:11]: Yes the println is intentional, because I wanted to have at least one std import.
